drivers/flash/spi_nor.h

```diff
@@ -33,29 +44,18 @@ struct spi_nor_config {
 #define SPI_NOR_CMD_WREN        0x06    /* Write enable */
 #define SPI_NOR_CMD_WRDI        0x04    /* Write disable */
 #define SPI_NOR_CMD_PP          0x02    /* Page program */
-#define SPI_NOR_CMD_SE          0x20    /* Sector erase */
-#define SPI_NOR_CMD_BE_32K      0x52    /* Block erase 32KB */
-#define SPI_NOR_CMD_BE          0xD8    /* Block erase */
 #define SPI_NOR_CMD_CE          0xC7    /* Chip erase */
```

Comment:
I'm intrigued that these seem to differ from what linux has for the opcodes in `include/linux/mtd/spi-nor.h`:

specifically that `0x20` was defined as `SPINOR_OP_BE_4K` and `0xD8` as `SPINOR_OP_SE`

Reply:
To some degree this is detritus.

What we're calling "JEDEC SPI-NOR" really isn't a standard; see commit comment of bb106390a29af81716ec89c0ae1be4e91dca69aa.  AFAIK none of these commands are formally standardized.

The M25P80 reference device, which Linux follows, uses 0xD8 for "sector erase" (sector = 512 KiBi = 64 KiBy), and 0xC7 for "bulk erase" (8 MiBi = 1 MiBy).  Those are the only erase commands supported in that device.  The concept of "block" in that device is limited to write protection of groups of sectors.

Later chips provided a smaller erasable unit (4 KiBy) and repurposed the term "sector" for that (probably based on disk file system terminology), calling the 64 KiBy unit a "block".  Even later, vendors supported a half-block (32 KiBy) unit.

The Winbond W25Q family was the primary target for the first instance of the driver in 87e54931670aabd6a6f69f773073d21cba72202c.  Its datasheet documents 0x20 as the "sector  erase" command where "sector" is 4 KiBy, and has "64 KiBy block erase" and "32 KiBy block erase" commands at 0xD8 and 0x52 respectively.  Generally devices have supported 4 KiBy and 64 KiBy erase commands, with some adding a 32 KiBy one.  Now we're finding devices that don't support any of these.

So that's history.  Much of the unending pain of maintaining this module has been related to slowly backing away from these hard-coded size/command associations.